### ISSUE TITLE
Change Request: Add additional RuleTester assertion option requireData

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#20390]
### ESLint version

10

### What problem do you want to solve?

The assertion options for the rule tester (issue #19921, [RFC](eslint/rfcs/pull/137)) allows requiring error properties for each invalid test case.
Currently there are options for requiring the `message` or `messageId` and the location properties `line`, `column`, `endLine` and `endColumn`.
The only property which cannot be required is `data` for checking the placeholders when using `messageId` (`suggestions` is required if there is one).

### What do you think is the correct solution?

Add an additional assertion option for requiring the `data` matcher: `requireData: boolean`.
This would require `data` iff an error matcher has a `messageId` property and the referenced message has placeholders.

For example given the referenced message "The function should have less than {{ max }} lines",  
the following would be valid:
```javascript
errors: ["message"]
errors: [{ message: "" }]
errors: [{}] // No messageId but would error as `message` or `messageId` is required
errors: [{ messageId: "id", data: { max: 1 } }]
errors: [{ messageId: "id", data: { max: 1, additional: true } }]

// These are valid for this assertion as the rule tester already reports "max" as missing
errors: [{ messageId: "id", data: {} }]
errors: [{ messageId: "id", data: { additional: true } }]
```

and the following would be invalid:
```
errors: 1
errors: [{ messageId: "id" }]
```

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by aladdin-add** [comment]
Overall, I support this proposal. 👍



> errors: 1

Semantically, it is equivalent to `errors: [{}]`, so I think it's more reasonable not to report an error in this case too. Alternatively, it would be even more reasonable to add another option to disallow number-form. thoughts?

**Comment by DMartens** [comment]
@aladdin-add I raised the same concern in the [original RFC](https://github.com/eslint/rfcs/pull/137#discussion_r2368322428) but the way I proposed is consistent with the existing assertion options. To disallow only the number form you can use `requireMessage: true`.

**Comment by mdjermanovic** [comment]
I support this proposal, but I also think `errors: 1` should be valid for this option regardless of its value because this option has an effect only when `messageId` is specified.

@eslint/eslint-team thoughts about this proposal?

**Comment by lumirlumir** [comment]
I also support this proposal and agree with the comments from @aladdin-add and @mdjermanovic. 

It seems more appropriate for `errors: 1` to be valid in this case.

**Comment by SwetaTanwar** [comment]
I'm also in favor of this proposal & agree that `errors: 1` should be valid. Marking as accepted.
